### PR TITLE
Issue#35

### PR DESCRIPTION
**Pull request by Netravati1205**
PR for Conditional Table Trigger Points

[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/14116658/Verifier_Test_Report.txt)

### CODE REVIEW

**Comment by Netravati1205** [comment]
I see there is an conflict I will resolve that

**Review by robe070 — CHANGES_REQUESTED** [review]
This is about conditional tests. So you need a test that matches the condition and one that does not match the condition. You only have the former.

Keeping your existing structure, you will need to duplicate each method that checks like mthCheck_Insert_Trigger. No messages will be expected back so that part is simplified.

File: Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml [hunk 1]
```diff
@@ -420,40 +497,100 @@
...
+    Endroutine
+    
+    
+    
+    *
+    *
+    *
+    Mthroutine Name(mthTestUnMatchTriggers)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+
```
Comment (robe070) [inline Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml:521]: Log is very plain with little info
```
 2024-02-02 00:18:57 -  Insert Trigger
    2024-02-02 00:18:57 -  - Successful
    2024-02-02 00:18:57 -  Update Trigger
    2024-02-02 00:18:57 -  - Successful
    2024-02-02 00:18:57 -  Fetch Trigger
    2024-02-02 00:18:57 -  - Successful
    2024-02-02 00:18:57 -  Delete Trigger
    2024-02-02 00:18:57 -  - Successful
```

Issue a message (once off) stating you are looking for no triggers being fired and thus no messages.

And for **each check**, state that 0 messages were found.

**Review by robe070 — CHANGES_REQUESTED** [review]
You must revert the merge of debug/paas. This repo requires a straight commit graph. 
Rebase your branch on debug/paas.

Most of the tables are missing from the User List. If you use the User List you'll not omit one!

This is not well named. Its not all of the Fields in the table
```
* #XG_ALL includes all Fields in the Table
Group_By Name(#XG_ALL) Fields(#BIFN0700A #BIFA032A)
```

Looks like this is not used for anything useful. Delete it.
```
Def_List Name(#w_lDB) Fields(#XG_ALL) Counter(#LF_DB_Count) Type(*WORKING) Entrys(*MAX)
```

Basically looking good. Do you think that you have any further functionality to add?

File: Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml (hunk 1 above)
Comment (Netravati1205) [reply Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml:521]: Okay, I will improve this.

**Comment by Netravati1205** [comment]
> You must revert the merge of debug/paas. This repo requires a straight commit graph. Rebase your branch on debug/paas.
> 
> Most of the tables are missing from the User List. If you use the User List you'll not omit one!
> 
> This is not well named. Its not all of the Fields in the table
> 
> ```
> * #XG_ALL includes all Fields in the Table
> Group_By Name(#XG_ALL) Fields(#BIFN0700A #BIFA032A)
> ```
> 
> Looks like this is not used for anything useful. Delete it.
> 
> ```
> Def_List Name(#w_lDB) Fields(#XG_ALL) Counter(#LF_DB_Count) Type(*WORKING) Entrys(*MAX)
> ```
> 
> Basically looking good. Do you think that you have any further functionality to add?

Yes, I noticed all the fields are not added to  **Group_By Name(#XG_ALL) Fields(#BIFN0700A #BIFA032A)** , I will add missed fields.

**Comment by Netravati1205** [comment]
> > You must revert the merge of debug/paas. This repo requires a straight commit graph. Rebase your branch on debug/paas.
> > Most of the tables are missing from the User List. If you use the User List you'll not omit one!
> > This is not well named. Its not all of the Fields in the table
> > ```
> > * #XG_ALL includes all Fields in the Table
> > Group_By Name(#XG_ALL) Fields(#BIFN0700A #BIFA032A)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Looks like this is not used for anything useful. Delete it.
> > ```
> > Def_List Name(#w_lDB) Fields(#XG_ALL) Counter(#LF_DB_Count) Type(*WORKING) Entrys(*MAX)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Basically looking good. Do you think that you have any further functionality to add?
> 
> Yes, I noticed all the fields are not added to **Group_By Name(#XG_ALL) Fields(#BIFN0700A #BIFA032A)** , I will add missed fields.


All the trigger points and conditions I have applied for the table level triggers, this covers all the trigger points.
I have made the changes for Field level changes also will commit it once this is fine for you.

**Comment by robe070** [comment]
> will commit it once this is fine for you.

I'm just confirming that the current state is that I'm waiting for you to make requested changes before I do anything further with this PR.

**Comment by Netravati1205** [comment]
> > will commit it once this is fine for you.
> 
> I'm just confirming that the current state is that I'm waiting for you to make requested changes before I do anything further with this PR.

Okay, I working on the requested changes

File: Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml (hunk 1 above)
Comment (Netravati1205) [reply Partition (WBP)/Functions/VI0035H.VTI0035.fun.yml:521]: Changes are pushed for review

**Comment by robe070** [comment]
All looks good.
Need Field Level Triggers now.

**Comment by Netravati1205** [comment]
Field Level Triggers added similar to table level triggers. All the points are covered as part of description.
[Verifier_Test_Report_rule_and_triggers.txt](https://github.com/lansa/db-regression/files/14140313/Verifier_Test_Report_rule_and_triggers.txt)

**Comment by Netravati1205** [comment]
Field Level Triggers also completed. Let me know if they are good and anything pending here.Thanks

**Comment by robe070** [comment]
I'll get there when I can. I've looked at #36 first as I figured you'll need to make the most changes for my first time view of the ticket's code. And that turned out to be the case.

**Pull request merged**